quanto/tensor/core.py

```diff
@@ -248,7 +337,14 @@ def forward(ctx, base, itype, axis=None):
         scale = (rmax - rmin) / (qmax - qmin)
         zeropoint = torch.round(-rmin / scale).to(torch.int8)
         data = torch.clamp(torch.round((base - rmin) / scale), min=0, max=2**bits - 1).to(torch.int8)
+
+        # pack the weights only if the weights are unpacked
+        can_pack_weights = data.shape[0] // 32 * itype.bits != 0
```

Comment:
Maybe we can move this check inside the `can_pack_weights` inside the pack_weights function + add log if we didn't manage to pack the weights.

Reply:
Makes sense, done !
I am just unsure on how the warnings are dealt in quanto, I left a comment